// dataforseo-mcp-server/build/main/core/modules/dataforseo-labs/tools/google/competitor-research/google-domain-rank-overview.tool.js
import { z } from 'zod';
import { BaseTool } from '../../../../base.tool.js';
export class GoogleDomainRankOverviewTool extends BaseTool {
    client;
    constructor(client) {
        super(client);
        this.client = client;
    }
    getName() {
        return 'dataforseo_labs_google_domain_rank_overview';
    }
    getDescription() {
        return `This endpoint will provide you with ranking and traffic data from organic and paid search for the specified domain. You will be able to review the domain ranking distribution in SERPs as well as estimated monthly traffic volume for both organic and paid results.`;
    }
    getParams() {
        return {
            target: z.string().describe(`target domain`),
            location_name: z.string().default("United States").describe(`full name of the location
required field
in format "Country"
example:
United Kingdom`),
            language_code: z.string().default("en").describe(`language code
        required field
        example:
        en`),
            ignore_synonyms: z.boolean().default(true).describe(`ignore highly similar keywords, if set to true, results will be more accurate`)
        };
    }
    async handle(params) {
        try {
            const response = await this.client.makeRequest('/v3/dataforseo_labs/google/domain_rank_overview/live', 'POST', [{
                    target: params.target,
                    location_name: params.location_name,
                    language_code: params.language_code,
                    ignore_synonyms: params.ignore_synonyms
                }]);
            return this.validateAndFormatResponse(response);
        }
        catch (error) {
            return this.formatErrorResponse(error);
        }
    }
}
