// dataforseo-mcp-server/build/main/core/modules/business-data-api/tools/listings/business-listings-filters.tool.js
import { z } from 'zod';
import { BaseTool } from '../../../base.tool.js';
export class BusinessListingsFiltersTool extends BaseTool {
    client;
    static cache = null;
    static lastFetchTime = 0;
    static CACHE_TTL = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
    // Map of tool names to their corresponding filter paths in the API response
    static TOOL_TO_FILTER_MAP = {
        'business_data_business_listings_search': 'search',
        'business_data_business_listings_categories_aggregation': 'categories_aggregation'
    };
    constructor(client) {
        super(client);
        this.client = client;
    }
    getName() {
        return 'business_data_business_listings_filters';
    }
    getDescription() {
        return `Here you will find all the necessary information about filters that can be used with Business Data API business listings endpoints.

Please, keep in mind that filters are associated with a certain object in the result array, and should be specified accordingly.`;
    }
    supportOnlyFullResponse() {
        return true;
    }
    getParams() {
        return {
            tool: z.string().optional().describe('The name of the tool to get filters for')
        };
    }
    async fetchAndCacheFilters() {
        const now = Date.now();
        // Return cached data if it's still valid
        if (BusinessListingsFiltersTool.cache &&
            (now - BusinessListingsFiltersTool.lastFetchTime) < BusinessListingsFiltersTool.CACHE_TTL) {
            return BusinessListingsFiltersTool.cache;
        }
        // Fetch fresh data
        const response = await this.client.makeRequest('/v3/business_data/business_listings/available_filters', 'GET', null, true);
        this.validateResponseFull(response);
        // Transform the response into our cache format
        const filters = {};
        const result = response.tasks[0].result[0];
        // Process each tool's filters
        for (const [toolName, filterPath] of Object.entries(BusinessListingsFiltersTool.TOOL_TO_FILTER_MAP)) {
            if (result && result[filterPath]) {
                filters[toolName] = result[filterPath];
            }
        }
        // Update cache
        BusinessListingsFiltersTool.cache = filters;
        BusinessListingsFiltersTool.lastFetchTime = now;
        return filters;
    }
    async handle(params) {
        try {
            const filters = await this.fetchAndCacheFilters();
            if (!params.tool) {
                return this.formatResponse(filters);
            }
            const toolFilters = filters[params.tool];
            if (!toolFilters) {
                throw new Error(`No filters found for tool: ${params.tool}`);
            }
            return this.formatResponse(toolFilters);
        }
        catch (error) {
            return this.formatErrorResponse(error);
        }
    }
}
