// dataforseo-mcp-server/build/main/core/modules/domain-analytics/tools/technologies/domain-technologies.tool.js
import { z } from 'zod';
import { BaseTool } from '../../../base.tool.js';
export class DomainTechnologiesTool extends BaseTool {
    client;
    constructor(client) {
        super(client);
        this.client = client;
    }
    getName() {
        return 'domain_analytics_technologies_domain_technologies';
    }
    getDescription() {
        return `Using this endpoint you will get a list of technologies used in a particular domain`;
    }
    getParams() {
        return {
            target: z.string().describe(`target domain
required field
domain name of the website to analyze
Note: results will be returned for the specified domain only`)
        };
    }
    async handle(params) {
        try {
            const response = await this.client.makeRequest('/v3/domain_analytics/technologies/domain_technologies/live', 'POST', [{
                    target: params.target
                }]);
            return this.validateAndFormatResponse(response);
        }
        catch (error) {
            return this.formatErrorResponse(error);
        }
    }
}
